src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/SynthesizedSymbols/AnonymousType.DelegateTemplateSymbol.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed partial class AnonymousTypeManager
+    {
+        internal sealed class AnonymousDelegateTemplateSymbol : AnonymousTypeOrDelegateTemplateSymbol
+        {
+            private readonly ImmutableArray<Symbol> _members;
+
+            /// <summary>
+            /// True if any of the delegate parameter types or return type are
+            /// fixed types rather than type parameters.
+            /// </summary>
+            internal readonly bool HasFixedTypes;
+
+            internal AnonymousDelegateTemplateSymbol(
+                AnonymousTypeManager manager,
+                string name,
+                TypeSymbol objectType,
+                TypeSymbol intPtrType,
+                TypeSymbol? voidReturnTypeOpt,
+                int parameterCount,
+                RefKindVector refKinds)
+                : base(manager, Location.None) // Location is not needed since NameAndIndex is set explicitly below.
+            {
+                Debug.Assert(refKinds.IsNull || parameterCount == refKinds.Capacity - (voidReturnTypeOpt is { } ? 0 : 1));
```

Comment:
>voidReturnTypeOpt is { } [](http://example.com/codeflow?start=87&length=24)

nit: consider extracting the various null checks on `voidReturnTypeOpt` in this method to a local (`returnsVoid` or some such).

Reply:
I'll leave as is because this constructor was essentially just moved from `SynthesizedDelegateSymbol`.